### PR TITLE
[MM-20559] Migrate 'components/admin_console/reset_password_modal' mo…

### PR DESCRIPTION
**Pull request by kosgrz**
…dule to TypeScript and create tests

#### Summary
This PR migrates 'components/admin_console/reset_password_modal' module to TypeScript and creates tests.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13513

### CODE REVIEW

File: components/admin_console/reset_password_modal/reset_password_modal.tsx [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
+}
+
+type State = {
+    serverErrorNewPass: JSX.Element|null;
+    serverErrorCurrentPass: JSX.Element|string|null;
+}
 
-    static defaultProps = {
+type Props = {
+    user: User|null;
```
Comment (devinbinnie) [inline components/admin_console/reset_password_modal/reset_password_modal.tsx]: You should be able to use the `?` shorthand to define that this object is nullable.
```suggestion
    user?: User;
```

File: components/admin_console/reset_password_modal/reset_password_modal.tsx [hunk 2]
```diff
@@ -1,29 +1,54 @@
...
-export default class ResetPasswordModal extends React.Component {
-    static propTypes = {
-        user: PropTypes.object,
-        currentUserId: PropTypes.string.isRequired,
-        show: PropTypes.bool.isRequired,
-        onModalSubmit: PropTypes.func,
-        onModalDismissed: PropTypes.func,
-        passwordConfig: PropTypes.object,
-    };
+interface User {
```
Comment (devinbinnie) [inline components/admin_console/reset_password_modal/reset_password_modal.tsx]: We have a `UserProfile` in `mattermost-redux` that this component can consume, so you don't need to define this type.

**Comment by kosgrz** [comment]
Thanks for the review @devinbinnie !

File: components/admin_console/reset_password_modal/reset_password_modal.tsx (hunk 1 above)
Comment (kosgrz) [reply components/admin_console/reset_password_modal/reset_password_modal.tsx]: done

File: components/admin_console/reset_password_modal/reset_password_modal.tsx (hunk 2 above)
Comment (kosgrz) [reply components/admin_console/reset_password_modal/reset_password_modal.tsx]: done

File: components/admin_console/reset_password_modal/reset_password_modal.tsx [hunk 3]
```diff
@@ -1,29 +1,50 @@
...
+}
+
+type Props = {
+    user?: UserProfile;
+    currentUserId: string;
+    show: boolean;
+    onModalSubmit: (user?: UserProfile) => void;
+    onModalDismissed: () => void;
+    passwordConfig: PasswordConfig;
+    actions: Actions;
```
Comment (jespino) [inline components/admin_console/reset_password_modal/reset_password_modal.tsx]: Probably makes sense to embed directly here the `actions` definition as a struct, I don't see the point of split that if we are not going to use that in any other place.

Comment (kosgrz) [reply components/admin_console/reset_password_modal/reset_password_modal.tsx]: Done

**Pull request merged**